### PR TITLE
Format better and remove unnecessary borrows

### PR DESCRIPTION
**Pull request by benarmstead**
Title self explanatory

### CODE REVIEW

**Review by nikomatsakis — APPROVED** [review]
These changes look pretty good to me.

**Comment by benarmstead** [comment]
The CI seems to have brought up one or two errors, I will try and deal with them in the morning.

File: tests/chars.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ fn half_open_correctness() {
 
     let range = low..high;
     let mut chars: Vec<char> = range.into_par_iter().collect();
-    chars.sort();
+    chars.sort_unstable();
```
Comment (cuviper) [inline tests/chars.rs:11]: Weird, this shouldn't need sorting at all...

File: tests/chars.rs [hunk 2]
```diff
@@ -26,7 +26,7 @@ fn closed_correctness() {
 
     let range = low..=high;
     let mut chars: Vec<char> = range.into_par_iter().collect();
-    chars.sort();
+    chars.sort_unstable();
```
Comment (cuviper) [inline tests/chars.rs:29]: Same here, it should already be sorted.

**Pull request closed without merging**